subsys/net/ip/net_context.c

```diff
@@ -2213,6 +2228,22 @@ static int set_context_sndtimeo(struct net_context *context,
 #endif
 }
 
+static int set_context_rcvbuf(struct net_context *context,
+				const void *value, size_t len)
+{
+#if defined(CONFIG_NET_CONTEXT_RCVBUF)
+	if (len != sizeof(uint16_t)) {
```

Comment:
I think the expected option type should rather be `int` to be compatible with Linux. We can check here if the value fits within expected range and return error if not.